sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/APISpec.groovy

```diff
@@ -437,83 +555,47 @@ class APISpec extends Specification {
     }
 
     def enableSoftDelete() {
-        primaryServiceURL.setProperties(new StorageServiceProperties()
+        primaryServiceClient.setProperties(new StorageServiceProperties()
             .deleteRetentionPolicy(new RetentionPolicy().enabled(true).days(2)))
-        sleep(30000) // Wait for the policy to take effect.
+
+        sleepIfRecord(30000)
     }
 
     def disableSoftDelete() {
-        primaryServiceURL.setProperties(new StorageServiceProperties()
+        primaryServiceClient.setProperties(new StorageServiceProperties()
             .deleteRetentionPolicy(new RetentionPolicy().enabled(false)))
 
-        sleep(30000) // Wait for the policy to take effect.
+        sleepIfRecord(30000)
     }
 
-
-    /*
-    This method returns a stub of an HttpResponse. This is for when we want to test policies in isolation but don't care
-     about the status code, so we stub a response that always returns a given value for the status code. We never care
-     about the number or nature of interactions with this stub.
-     */
-
-    def getStubResponse(int code) {
-        return Stub(HttpResponse) {
-            statusCode() >> code
+    // Only sleep if test is running in live mode
+    def sleepIfRecord(long milliseconds) {
```

Comment:
Why you take million sec here? Can we take second?

Reply:
That is what `Thread.sleep` takes.